lib/api_repo.js

```diff
@@ -424,6 +424,49 @@ function collectProtoDeps(protoFile, includePath, done) {
 }
 
 /**
+ * Create the mapping of the .proto file source to the destination.
+ *
+ * This is a sub task of 'ProtoBased' nodeJS module in _buildProtos, but
+ * extracted as a method for the testing.
+ *
+ * @param {string} outDir - the top of the output directory.
+ * @param {[][]string} fileList - the list of the lists of the .proto files
+ *   to be copied.
+ * @returns {Object} - an object whose keys are the source files (i.e. items
+ *   in fileList) and values are the path of the files relative to
+ *   the output destinations.
+ */
+ApiRepo.prototype._buildProtoFilesMapping = function _buildProtoFilesMapping(
+    fileLists, includePath) {
+  var filesMap = {};
+  // Each item in fileLists is the list of .proto files calculated by
+  // collectProtoDeps.
+  fileLists.forEach(function(fileList) {
+    fileList.forEach(function(filePath) {
+      if (filePath in filesMap) {
+        return;
+      }
+      for (var i = 0; i < includePath.length; ++i) {
+        if (filePath.indexOf(includePath[i]) === 0) {
```

Comment:
google.protobuf files are typically in the default installation path of protobuf package itself (i.e. /usr/include, for example), and that does not appear in the includePath (these are additional paths for protoc).
Also this path itself relies on the location of protoc command. For example, in my case this is actually ~/.linuxbrew/include.  To deal with this situation, I think it's better to special-handle the google/protobuf/*.proto files (and do it after trying includePath, to avoid duplicated names).

Added a bit more comments to explain this.